### PR TITLE
bpo-44640: Improve punctuation in isinstance() error message

### PR DESCRIPTION
**Pull request by wyz23x2**
Unify the punctuation marks of `isinstance()` and `issubclass()` error messages.

https://bugs.python.org/issue44640

### CODE REVIEW

**Comment by wyz23x2** [comment]
Does this need news? IMO it's a very minor change, but it involves frequently-used builtins.

**Comment by jdevries3133** [comment]
The word "union" is actually an exception to the "a/an" rule. Here is more detail:

https://english.stackexchange.com/questions/266309/why-is-union-an-exception-to-the-a-an-rule

**Comment by wyz23x2** [comment]
OK :) English isn't my first language.

**Comment by jdevries3133** [comment]
@wyz23x2 It's all good! English is very fickle and the fact that "union" is an exception to the a/an rule is a pretty obscure one.

To be clear, I don't think that the punctuation removal is necessary either; the original message is a grammatically correct full sentence. In its current state, you left behind the first comma, and removed all the following punctuation, which is not correct.

Unfortunately, as far as I can tell, there are no changes needed here; but feel free to let me know if you think I'm missing something.

**Comment by wyz23x2** [comment]
I agree it's somehow a "micro" change and maybe not needed.
> ..., and removed all the following punctuation, which is not correct.

I just thought it's strange because all others aren't really full sentences since they have no periods:
```python
>>> import math, sys
>>> int([])
Traceback (most recent call last):
  ...
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'list'
>>> math.sqrt(-1)
Traceback (most recent call last):
  ...
ValueError: math domain error
>>> [3.14][sys.maxsize+100]
Traceback (most recent call last):
  ...
IndexError: cannot fit 'int' into an index-sized integer
>>> import mathh
Traceback (most recent call last):
  ...
ModuleNotFoundError: No module named 'mathh'
```
etc.
Changing all of them would require a quite big rewrite of the test suite (and I think there's _lots_ of them in the stdlib!).

**Comment by jdevries3133** [comment]
But then again, those other messages are simpler single-statement messages. Since this message is a three-part list, maybe the punctuation in this case provides some needed clarity?

In any case, you have to think about those chaotic individuals who have written code which asserts against the content of the error message here. Python is big enough that it probably has happened, and this change will break that person's (albeit questionable) code.

File: Objects/abstract.c
Comment (ambv) [inline Objects/abstract.c]: The comma before "or" is an Oxford comma and is correct for this use. It improves clarity that "or a union" refers to "arg 2" and not to "a tuple of".

Let's keep the Oxford comma. Please add it to the other error message for consistency.

**Comment by Fidget-Spinner** [comment]
> In any case, you have to think about those chaotic individuals who have written code which asserts against the content of the error message here. Python is big enough that it probably has happened, and this change will break that person's (albeit questionable) code.

It's unlikely for there to be much code reliant on that. Union support is 3.10 onwards only, and 3.10.0 isn't out yet (only the betas). So I don't think any code is going to break.

File: Objects/abstract.c
Comment (wyz23x2) [reply Objects/abstract.c]: OK. Thanks!

**Comment by wyz23x2** [comment]
> But then again, those other messages are simpler single-statement messages. Since this message is a three-part list, maybe the punctuation in this case provides some needed clarity?

While, the `int()` error message is also similar format, but doesn't have a comma before `or` (but maybe because of `, not blah` after it), and doesn't end with a period.

**Comment by wyz23x2** [comment]
P.S. I'm sure that it was noted before, but actually _any_ object that implements `__int__`, `__trunc__` or `__index__` is supported, not just the ones specified in the message.

**Review by Fidget-Spinner — APPROVED** [review]
LGTM. I'll leave it up to the core dev merging this to decide whether we should backport this to 3.10 or not (personally, I don't think it will hurt, but this doesn't look like a something we usually backport either so I'm not too sure).

**Comment by jacobtylerwalls** [comment]
Without offering an opinion on whether this change is desirable, just offering that you could say "Add serial comma" rather than "Fix punctuation". Yes, a period is added also, but if this is change is worth merging, it would likely only be for the serial comma.

**Comment by wyz23x2** [comment]
Ping.

**Pull request merged**
